### PR TITLE
Navigation update separate mobile desktop

### PR DESCRIPTION
**Pull request by JesJehle**
I made a first version of a separate navigation for mobile and desktop.
The mobile navigation is simple and I used the pages and routes for the map, list, settings and tree details.
The desktop navigation is more complicated and not finished yet. Since we are only working with one page and additional popovers and grids, I created a separate desktop page. 
Here I used a grid to display the inventory list and map components, which works after some trial and error. 
Now comes the more complicated part. 
At the moment settings and treedetails are implemented as pages, to show them in a popup window next to the inventory list, I need to get the routing logic into some kind of state to render the clicked treedetails. 
A short talk would be helpful here.
I would rather not merge the current branch, just yes, but please have a look and give feedback. I would implement the changes for the desktop version first and then merge.

### CODE REVIEW

**Comment by mmaelicke** [comment]
I added the logic, how the navigation would be used within a single page to show different component. This is also not really working. It feels like we are extremely fighting the framework, trying to do stuff the components were not made for.

I have to investigate a bit more, why the `match` object supplied by the Router is not updating appropriately. 

I am not sure, how we would implement navigation to tree details in mobile, but load the components directly on Desktop. 

My last shot are nested router-outlets, which gave me really weird animations in the first place. If that does not work, I guess we need to take a completely different route.

**Comment by mmaelicke** [comment]
check https://github.com/hydrocode-de/mathislewald/tree/use-nested-router 

this one implements the nested router. Works, kind of, but has a annoying delay.

**Comment by mmaelicke** [comment]
Got it. Works with the `react-router` directly. The IonRouter is made for *Page*transitions, that screwed everything up.
Working with `react-router` directly is possible, but we need to stick to version 5.2 as the new version (6.X) is not yet supported by ionic.

**Comment by JesJehle** [comment]
Since, the initial draft is done, I would suggest merging this branch to main and continuing from there.
Feel free to merge or request changes.

**Comment by JesJehle** [comment]
To Do:
- [ ] Merge #48 with update-list
- [ ] Create selectedVariable state in context #51 
- [ ] Replace IonGrid with div in DesktopPage #50 
- [ ] Create ButtonGroup component
- [ ] Add Button Group to Mobile and Desktop pages
- [ ] Update Visual Map Selection with new Button
- [ ] Add Filter Popover to DesktopPage
- [ ] Add Filter modal to MobilePage
- [ ] Add Variable Selection Popover to DesktopPage
- [ ] Add Variable Selection Modal to MobilePage

**Comment by JesJehle** [comment]
Ok, I think I am done. @mmaelicke have a look, merge if possible.

File: src/components/ActiveMapSelectionButton.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      setSrc("assets/dtm.png");
+      setTitel("DTM");
+    } else if (layers.activeBaseLayer.toString() === "ortho") {
+      setSrc("assets/ortho.png");
+      setTitel("Ortho");
+    } else {
+      setSrc("assets/openstreetmap.png");
+      setTitel("OSM");
+    }
+  }, [layers.activeBaseLayer]);
```
Comment (mmaelicke) [inline src/components/ActiveMapSelectionButton.tsx:33]: a comment about the purpose would be nice

File: src/components/InventoryList.tsx [hunk 2]
```diff
@@ -131,6 +135,36 @@ export default InventoryList;
...
+//         </IonCardTitle>
+//       </IonCol>
+//     ) : null}
+//     <IonCol class="ion-text-center ion-hide-sm-down">
+//       <IonCardSubtitle>RADIUS</IonCardSubtitle>
+//       <IonCardTitle>{f.properties.radius.toFixed(1)}</IonCardTitle>
+//     </IonCol>
+//     <IonCol class="ion-text-end">
+//       <IonButton
+//         // fill="outline"
```
Comment (mmaelicke) [inline src/components/InventoryList.tsx:159]: let's clean this up.

File: src/components/MainMapMaplibre.tsx [hunk 3]
```diff
@@ -74,6 +74,7 @@ const MainMap: React.FC = () => {
         // 47.884438269626294, 8.088652498339387
       }}
     >
+      <div></div>
```
Comment (mmaelicke) [inline src/components/MainMapMaplibre.tsx:77]: what is this div for?

File: src/components/MapSelectionButton.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import { IonButton, IonImg, IonNote } from "@ionic/react";
+import React from "react";
+import { useLayers } from "../context/layers";
+
+interface MapSelectionButtonProps {
+  name: string;
+  titel: string;
```
Comment (mmaelicke) [inline src/components/MapSelectionButton.tsx:7]: can we use `title` ?

File: src/components/VariableSelector.tsx
Comment (mmaelicke) [inline src/components/VariableSelector.tsx:1]: This component doesn't do anything up to now, right?

**Review by mmaelicke — APPROVED** [review]
Really nice work!

I added one commit to disable the `Selection` button for now. I will approve, although there are some minor comments. Mainly about re-naming and commenting.

File: src/components/MapSelectionButton.tsx (hunk 4 above)
Comment (JesJehle) [reply src/components/MapSelectionButton.tsx:7]: sure

File: src/components/VariableSelector.tsx
Comment (JesJehle) [reply src/components/VariableSelector.tsx:1]: No, this one is used in `VariableSelectionPopover` and `VariableSelectionModal`.

**Comment by JesJehle** [comment]
Ok, I resolved all issues you pointed out. I will merge this now.

**Pull request merged**